### PR TITLE
[ci] add a script to get the workflow job id

### PR DESCRIPTION
**Pull request by suo**
This is inexplicably not available in the runtime environment of a job,
so we need to retrieve it using the GH API.

### CODE REVIEW

File: .github/scripts/get_workflow_job_id.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+PYTORCH_REPO = "https://api.github.com/repos/pytorch/pytorch"
+GITHUB_TOKEN = os.environ["GITHUB_TOKEN"]
+REQUEST_HEADERS = {
+    "Accept": "application/vnd.github.v3+json",
+    "Authorization": "token " + GITHUB_TOKEN,
+}
+
+response = requests.get(
+    # No f-strings because our CI needs to be able to run on older Python versions
+    PYTORCH_REPO + "/actions/runs/" + args.workflow_run_id + "/jobs?per_page=100",
```
Comment (janeyx99) [inline .github/scripts/get_workflow_job_id.py:43]: this means that we are limited on 100 jobs per workflow correct? just for my understanding, as i think that’s a reasonable requirement anyway

**Review by janeyx99 — APPROVED** [review]
awesome! thanks for looking into this!

File: .github/scripts/get_workflow_job_id.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import requests
+import os
+import argparse
+
+# Our strategy is to retrieve the parent workflow run, then filter its jobs on
+# RUNNER_NAME to figure out which job we're currently running.
+#
+# Why RUNNER_NAME? Because it's the only thing that uniquely identifies a job within a workflow.
+# GITHUB_JOB doesn't work, as it corresponds to the job yaml id
+# (https://bit.ly/37e78oI), which is has two problems:
```
Comment (janeyx99) [inline .github/scripts/get_workflow_job_id.py:14]: nit
```suggestion
# (https://bit.ly/37e78oI), which has two problems:
```

File: .github/scripts/get_workflow_job_id.py (hunk 1 above)
Comment (suo) [reply .github/scripts/get_workflow_job_id.py:43]: yes. Actually maybe I should just fix that—when we switch to consolidated workflows it's plausible that we get more than 100

**Pull request closed without merging**